packages/taiko-client/proposer/proposer.go

```diff
@@ -537,40 +668,84 @@ func (p *Proposer) Name() string {
 
 // Profitability is determined by comparing the revenue from transaction fees
 // to the costs of proposing and proving the block. Specifically:
+func (p *Proposer) isProfitable(txLists []types.Transactions, proposingCosts *big.Int) (bool, error) {
+	totalTransactionFees, err := p.calculateTotalL2TransactionsFees(txLists)
+	if err != nil {
+		return false, err
+	}
+
+	costs, err := p.estimateTotalCosts(proposingCosts)
+	if err != nil {
+		return false, err
+	}
+
+	log.Debug("isProfitable", "total L2 fees", totalTransactionFees, "total L1 costs", costs)
+
+	return totalTransactionFees.Cmp(costs) > 0, nil
+}
 
-func (p *Proposer) isProfitable(txList types.Transactions) (bool, error) {
+func (p *Proposer) calculateTotalL2TransactionsFees(txLists []types.Transactions) (*big.Int, error) {
 	totalTransactionFees := new(big.Int)
-	totalGasConsumed := uint64(0)
+	totalGasConsumed := new(big.Int)
 
-	for _, tx := range txList {
-		// TODO: Not sure if this is the best approach here.
-		// Maybe txList.EstimatedGasUsed is useful? Maybe we need to calculate it ourselves?
-		gasConsumed := tx.Gas()
-		priorityGasPrice, err := p.getPriorityGasPrice(tx)
-		if err != nil {
-			return false, err
+	baseL2Fee, err := p.rpc.L2.SuggestGasPrice(p.ctx)
+	if err != nil {
+		return nil, err
+	}
+	priorityL2Fee, err := p.rpc.L2.SuggestGasTipCap(p.ctx)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
These should come from the l2 tx not from the L2 rpc

Reply:
Fixed